src/core/config/__tests__/CustomModesManager.spec.ts

```diff
@@ -436,835 +445,848 @@ describe("CustomModesManager", () => {
-					recursive: true,
-				})
-				expect(fs.mkdir).toHaveBeenCalledWith(
-					expect.stringContaining(path.join("rules-imported-mode", "subfolder")),
-					{ recursive: true },
-				)
+				const modes = await manager.getCustomModes()
 
-				// Verify file contents
-				const rule1Path = Object.keys(writtenFiles).find((p) => p.includes("rule1.md"))
-				const rule2Path = Object.keys(writtenFiles).find((p) => p.includes("rule2.md"))
-				expect(writtenFiles[rule1Path!]).toBe("Rule 1 content")
-				expect(writtenFiles[rule2Path!]).toBe("Rule 2 content")
+				// Should continue without the directory that couldn't be read
+				expect(modes).toHaveLength(0)
 			})
 
-			it("should import multiple modes at once", async () => {
-				const importYaml = yaml.stringify({
-					customModes: [
-						{
-							slug: "mode1",
-							name: "Mode 1",
-							roleDefinition: "Role 1",
-							groups: ["read"],
-						},
-						{
-							slug: "mode2",
-							name: "Mode 2",
-							roleDefinition: "Role 2",
-							groups: ["edit"],
-							rulesFiles: [
-								{
-									relativePath: "rules-mode2/rule.md",
-									content: "Mode 2 rules",
-								},
-							],
-						},
-					],
-				})
+			it("should handle concurrent file operations safely", async () => {
+				const mode1: ModeConfig = {
+					slug: "concurrent-1",
+					name: "Concurrent 1",
+					roleDefinition: "Role 1",
+					groups: ["read"],
+					source: "global",
+				}
+				const mode2: ModeConfig = {
+					slug: "concurrent-2",
+					name: "Concurrent 2",
+					roleDefinition: "Role 2",
+					groups: ["read"],
+					source: "global",
+				}
+				const mode3: ModeConfig = {
+					slug: "concurrent-3",
+					name: "Concurrent 3",
+					roleDefinition: "Role 3",
+					groups: ["read"],
+					source: "global",
+				}
 
-				let roomodesContent: any = null
+				let settingsContent = { customModes: [] }
 				;(fs.readFile as Mock).mockImplementation(async (path: string) => {
 					if (path === mockSettingsPath) {
-						return yaml.stringify({ customModes: [] })
-					}
-					if (path === mockRoomodes && roomodesContent) {
-						return yaml.stringify(roomodesContent)
+						// Simulate delay to test race conditions
+						await new Promise((resolve) => setTimeout(resolve, Math.random() * 10))
+						return yaml.stringify(settingsContent)
 					}
 					throw new Error("File not found")
 				})
 				;(fs.writeFile as Mock).mockImplementation(async (path: string, content: string) => {
-					if (path === mockRoomodes) {
-						roomodesContent = yaml.parse(content)
-					}
-					return Promise.resolve()
-				})
-
-				const result = await manager.importModeWithRules(importYaml)
-
-				expect(result.success).toBe(true)
-				expect(roomodesContent.customModes).toHaveLength(2)
-				expect(roomodesContent.customModes[0].slug).toBe("mode1")
-				expect(roomodesContent.customModes[1].slug).toBe("mode2")
-			})
-
-			it("should handle import errors gracefully", async () => {
```

Comment:
@roomote-agent this test was removed, should it be like that? is this still being validated elsewhere?